protocols/imap/src/test/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParserTest.java

```diff
@@ -0,0 +1,243 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.decode.parser;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+
+import org.apache.james.imap.api.ImapCommand;
+import org.apache.james.imap.decode.ImapRequestStreamLineReader;
+import org.apache.james.imap.message.request.GetAnnotationRequest;
+import org.apache.james.imap.message.request.GetAnnotationRequest.Depth;
+import org.apache.james.protocols.imap.DecodingException;
+import org.junit.Test;
+
+public class GetAnnotationCommandParserTest {
+
+    private static final String INBOX = "anyInboxName";
+    private static final String TAG = "A1";
+    private GetAnnotationCommandParser parser = new GetAnnotationCommandParser();
+    private ImapCommand command = ImapCommand.anyStateCommand("Command");
+
+    @Test(expected = DecodingException.class)
+    public void decodeMessageShouldThrowsExceptionWhenCommandHasNotMailbox() throws DecodingException {
+        InputStream inputStream = new ByteArrayInputStream(" \n".getBytes());
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
+
+        parser.decode(command, lineReader, TAG, null);
+    }
+
+    @Test
+    public void decodeMessageShouldReturnRequestWhenCommandHasMailboxOnly() throws DecodingException {
+        InputStream inputStream = new ByteArrayInputStream((INBOX + "    \n").getBytes());
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
+
+        GetAnnotationRequest request = (GetAnnotationRequest) parser.decode(command, lineReader, TAG, null);
+
+        assertThat(request.getMailboxName()).isEqualTo(INBOX);
+        assertThat(request.getKeys()).isEmpty();
+    }
+
+    @Test(expected = DecodingException.class)
+    public void decodeMessageShouldThrowExceptionWhenCommandHasOneKeyButInWrongFormat() throws DecodingException {
+        InputStream inputStream = new ByteArrayInputStream((INBOX + " /private/comment extrastring \n").getBytes());
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
+
+        parser.decode(command, lineReader, TAG, null);
+    }
+
+    @Test
+    public void decodeMessageShouldReturnRequestWhenCommandHasOnlyOneKey() throws DecodingException {
+        InputStream inputStream = new ByteArrayInputStream((INBOX + " /private/comment \n").getBytes());
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
```

Comment:
Extract it in a constant to give it a name.

I'm not supposed to guess it from github, where we do the review, and I don't want to open my IDE to find it out...